test/functional/services/doc_table.js

```diff
@@ -54,7 +55,12 @@ export function DocTableProvider({ getService }) {
 
     async toggleRowExpanded(row) {
       const rowExpandToggle = await this.getRowExpandToggle(row);
-      return await rowExpandToggle.click();
+      await rowExpandToggle.click();
+
+      const detailsRow = await row.findByXpath('./following-sibling::tr');
+      return await retry.try(async () => {
+        return detailsRow.findByCssSelector('doc-viewer');
```

Comment:
The XPath expression and the `doc-viewer` selector strongly couple this code to the dom structure. Maybe using `data-test-subj` for those would be a good idea?

Reply:
Updated https://github.com/elastic/kibana/pull/15203/commits/cf46bc1bc5bf4fa57aa17df0f3381401186aac67